File: src/main/java/me/austin/queer/feature/command/Command.java
package me.austin.queer.feature.command;

import com.mojang.brigadier.CommandDispatcher;

import me.austin.queer.feature.Feature;
import net.minecraft.server.command.ServerCommandSource;

public abstract class Command extends Feature {
    public final String syntax;

    public Command(String name, String syntax) {
        super(name);
        
        this.syntax = syntax;
    }

    public abstract void execute(String[] args);

    public abstract void register(CommandDispatcher<ServerCommandSource> dispatcher);
}

File: src/main/kotlin/me/austin/queer/commands/Command.kt
package me.austin.queer.commands

import com.mojang.brigadier.CommandDispatcher

import net.minecraft.server.command.ServerCommandSource
import me.austin.queer.feature.Nameable

abstract class Command(name: String, val syntax: String) : Nameable(name) {
    abstract fun execute(args: Array<String>)

    abstract fun register(dispatcher: CommandDispatcher<ServerCommandSource>)
}